vendor/github.com/hashicorp/vault/sdk/helper/parseutil/parseutil.go

```diff
@@ -67,6 +67,54 @@ func ParseDurationSecond(in interface{}) (time.Duration, error) {
 	return dur, nil
 }
 
+func ParseAbsoluteTime(in interface{}) (time.Time, error) {
+	var t time.Time
+	switch inp := in.(type) {
+	case nil:
+		// return default of zero
+		return t, nil
+	case string:
+		// Allow RFC3339 with nanoseconds, or without,
+		// or an epoch time as an integer.
+		var err error
+		t, err = time.Parse(time.RFC3339Nano, inp)
+		if err == nil {
+			break
+		}
+		t, err = time.Parse(time.RFC3339, inp)
+		if err == nil {
+			break
+		}
+		epochTime, err := strconv.ParseInt(inp, 10, 64)
+		if err == nil {
+			t = time.Unix(epochTime, 0)
+			break
+		}
+		return t, errors.New("could not parse string as date and time")
+	case json.Number:
+		epochTime, err := inp.Int64()
+		if err != nil {
+			return t, err
+		}
+		t = time.Unix(epochTime, 0)
+	case int:
```

Comment:
nitpick: you could have something like `case int, int32, uint, uint32, uint64` instead of so many switch cases doing the same thing?

also i guess the other weird thing is going from something close to a maxed out `uint64` would overflow an `int64`, but that's also not really a realistic time

Reply:
ignore this, didn't notice it was already merged